Don't retain context in locker

klauspost opened the pull request:
## Description

Send context as the parameter for locking instead of keeping it in the locker.

Makes it a bit more flexible and obvious where the context is being applied and generally considered best practice.

## How to test this PR?

Purely mechanical refactor

## Types of changes
- [x] Refactor

klauspost committed:
Don't retain context in locker

Send context as the parameter instead of keeping it in the locker.

Makes it a bit more obvious where the context is being applied and generally considered best practice.

klauspost committed:
Merge branch 'master' into dont-retain-context-in-locker

# Conflicts:
#	cmd/erasure-zones.go
#	cmd/server-main.go

klauspost committed:
Fix problem after merge.

harshavardhana commented on cmd/disk-cache-backend.go:
```diff
@@ -175,8 +175,8 @@ func newDiskCache(ctx context.Context, dir string, config cache.Config) (*diskCa
-	cache.NewNSLockFn = func(ctx context.Context, cachePath string) RWLocker {
-		return cache.nsMutex.NewNSLock(ctx, nil, cachePath, "")
+	cache.NewNSLockFn = func(cachePath string) RWLocker {
+		return cache.nsMutex.NewNSLock(nil, cachePath, "")
```
Here it should new NewNSLock without the `nil`?

klauspost replied:
Not sure I understand. TBH I don't understand the original code, but this should do the same AFAICT?

harshavardhana replied:
What I am saying is that if you remove everywhere, not remove this for lsync as well @klauspost consistently changed across.

klauspost replied:
Still not sure what you mean. The `nil` is a ` lockersFn func() []dsync.NetLocker` - it is not a context.

klauspost committed:
Merge branch 'master' into dont-retain-context-in-locker

# Conflicts:
#	cmd/data-crawler.go
#	cmd/disk-cache-backend.go
#	cmd/erasure-object.go
#	cmd/erasure-server-sets.go
#	cmd/iam.go
#	cmd/namespace-lock.go
#	cmd/server-main.go

klauspost committed:
Merge branch 'master' into dont-retain-context-in-locker

klauspost committed:
Fix leaked locks on cancelled contexts

Use the context for internal timeouts, but disconnect it from outgoing calls so we always receive the results and cancel it remotely.
